WalletWasabi/WabiSabi/Client/AmountDecomposer.cs

```diff
@@ -280,12 +296,29 @@ public IEnumerable<Money> Decompose(IEnumerable<Money> myInputCoinEffectiveValue
 					count,
 					stdDenoms).Select(Money.Satoshis).ToList();
 
+				// Translate back to denominations.
+				List<Output> finalDenoms = new();
+				foreach (var outputPlusFee in currentSet)
+				{
+					finalDenoms.Add(denoms.First(d => d.EffectiveCost == outputPlusFee));
+				}
+
+				// The decomposer won't take vsize into account for different script types, checking it back here if too much, disregard the decomposition.
+				var totalVSize = finalDenoms.Sum(d => d.ScriptType.EstimateOutputVsize());
+				if (totalVSize > AvailableVsize)
+				{
+					continue;
+				}
```

Comment:
This shouldn't be possible if `maxNumberOfOutputsAllowed` is calculated pessimistically instead of optimistically.

Reply:
But it is calculated optimistically (that word exists? :)).

We have a difference in mindset here. Pessimistic throwing away good solutions. Optimistic will select the wrong solutions, which can be filtered later.  

It is possible to have a good candidate with mixed scripts that are below the available vsize and the alg should provide those.

Reply:
ok.